src/NServiceBus.FileBasedRouting/XmlRoutingFileAccess.cs

```diff
@@ -1,27 +1,23 @@
 ﻿namespace NServiceBus.FileBasedRouting
 {
     using System;
-    using System.IO;
     using System.Xml;
     using System.Xml.Linq;
 
     class XmlRoutingFileAccess
     {
-        public string FilePath { get; }
+        public Uri FileUri { get; }
 
-        public XmlRoutingFileAccess(string filePath)
+        public XmlRoutingFileAccess(Uri fileUri)
         {
-            FilePath = filePath;
+            FileUri = fileUri;
         }
 
         public XDocument Read()
         {
             try
             {
-                using (var fileStream = File.OpenRead(FilePath))
-                {
-                    return XDocument.Load(fileStream);
-                }
+                return XDocument.Load(FileUri.ToString());
```

Comment:
@Scooletz so the SO thread states to set the XmlResolver to `null` to not load external documents. but the official documentations states:

> Starting with the .NET Framework 4.5.2, this setting has a default value of null.

as we're compiling against 4.6, we should be good regarding an XXE vulnerability?